public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -504,19 +508,34 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
             boolean isMeasurement = false;
 
             List<AchillesAnalysis> analysisList = achillesAnalysisDao.findConceptAnalysisResults(conceptId,analysisIds);
+            HashMap<Long, AchillesAnalysis> analysisHashMap = new HashMap<>();
+            for(AchillesAnalysis aa: analysisList){
+                    analysisHashMap.put(aa.getAnalysisId(), aa);
+            }
 
-            conceptAnalysis.setConceptId(conceptId);
+            AchillesAnalysis analysis = analysisHashMap.get(COUNT_ANALYSIS_ID);
+            String unitName = null;
+            if(analysis != null){
+                for(AchillesResult results: analysis.getResults()){
+                    unitName = results.getStratum4();
+                }
+            }
 
-            for(AchillesAnalysis aa:analysisList){
-                    if(aa.getAnalysisId() == GENDER_ANALYSIS_ID){
+            conceptAnalysis.setConceptId(conceptId);
+            Iterator it = analysisHashMap.entrySet().iterator();
+            while(it.hasNext()) {
+                    Map.Entry pair = (Map.Entry)it.next();
+                    Long analysisId = (Long)pair.getKey();
+                    AchillesAnalysis aa = (AchillesAnalysis)pair.getValue();
```

Comment:
Just set unitName on every analysis right here. It doesn't matter if it is null. And make your copy constructor use it.

Reply:
Done.